### DISCUSSION TITLE
Get rid of isodate

### DISCUSSION THREAD

**Discussion by aucampia** [RDFLib/rdflib#2419] [TO-DO list]
isodate is abandonware, we need to find a replacement.

**Comment by ghost** [comment]
> isodate is abandonware, we need to find a replacement.

Maybe https://github.com/metomi/isodatetime?

**Reply by aucampia** [reply]
That looks quite good actually, but it does not seem that popular and I would be a bit worried that it also gets abandoned somewhere down the line.

Another option may be https://docs.astropy.org/en/stable/time/index.html - but that would only work if it is optional as it is part of astropy which is quite heavy.

**Comment by nicholascar** [comment]
We don't need isodate in Python 3.11+ so when we release RDFLib 8.0.0 which will require 3.21+, we will remove the isodate dependency. RDFLib 7.x will always depend on it and it was updated recently so things will be stable there with static versions of RDFLib.

**Reply by ajnelson-nist** [reply]
> ... so when we release RDFLib 8.0.0 which will require 3.21+, ...

Double-checking which way the typo went - did you mean will require 3.11+, or 3.12+?

**Reply by ashleysommer** [reply]
@nicholascar My understanding is we are targeting Python 3.9+ for RDFLib 8.
